src/index.js

```diff
@@ -1,567 +1,363 @@
-      public: buf2hex(auth.publicKey.reverse()),
+      public: auth_pub,
+      address: addressFromNetworkPublic(auth_pub)
     }
-  };
-};
-
-
-
-/**
- * Perform a sanity check on a set of three shards, checking that any two of
- * them can be combined to produce a reference value.
- * @param  {array of strings => string}  combiner a shard-combining function
- * @param  {string}  ref a reference value for comparison
- * @param  {string => string => bool}  comp a comparison function
- * @param  {array of strings}  shards an array of shards
- * @return  {bool}  true if shards consistent
- */
-const shardsConsistent = (combiner, ref, comp, shards) => {
-  const set0 = shards.slice(0, 2);
-  const set1 = shards.slice(1);
-  const set2 = [shards[0], shards[2]];
-
-  const check0 = comp(combiner(set0), ref);
-  const check1 = comp(combiner(set1), ref);
-  const check2 = comp(combiner(set2), ref);
-
-  return check0 && check1 && check2;
-}
-
-
-
-/**
- * Reduce a collection of arrays by recursive applications of bytewise XOR.
- * @param  {Array of Array of integers}  arrays an array of arrays
- * @return {Array} the resulting array
- */
-const reduceByXor = (arrays) => {
-  return arrays.reduce((acc, arr) =>
-    lodash.zipWith(acc, arr, (x, y) => x ^ y));
-}
-
-
-
-/**
- * Encode a hex string as three shards, such that any two shards can be
- * combined to recover it.
- *
- * @param  {string}  string hex-encoded string
- * @return {Array of strings} resulting shards
- */
-const shardHex = hex => {
-  const buffer = hex2buf(hex);
-  const sharded = shardBuffer(buffer);
-  const shards = sharded.map(pair =>
-    lodash.reduce(pair, (acc, arr) =>
-      acc + buf2hex(Buffer.from(arr)), ''));
-
-  if (shardsConsistent(combineHex, hex, lodash.isEqual, shards)) {
-    return shards;
-  } else {
-    // shards must always be consistent, so this should be unreachable
-    /* istanbul ignore next */
-    throw("shardHex: inconsistent shards -- please report this as a bug");
   }
 }
 
 
 
 /**
- * Encode a @q-encoded string as three shards, such that any two shards can be
- * combined to recover it.
- *
- * @param  {string}  string @q-encoded string
- * @return {Array of strings} resulting shards
+ * Convert a hex-encoded secp256k1 public key into an Ethereum address.
+ * @param  {String}  pub a (compressed) hex-encoded public key
+ * @return  {String}  the corresponding Ethereum address
  */
-const shardPatq = patq => {
-  const hexed = shardHex(ob.patq2hex(patq))
-  const shards = hexed.map(ob.hex2patq)
-
-  if (shardsConsistent(combinePatq, patq, ob.eqPatq, shards)) {
-    return shards;
-  } else {
-    // shards must always be consistent, so this should be unreachable
-    /* istanbul ignore next */
-    throw("shardPatq: inconsistent shards -- please report this as a bug");
-  }
+const addressFromSecp256k1Public = pub => {
+  const hashed = util.keccak256(Buffer.from(pub, 'hex'))
+  const addr = util.addHexPrefix(hashed.slice(12).toString('hex'))
```

Comment:
The `.slice(12)` reads a bit weird here, might want to put a comment to clarify that this is taking the last 20 bytes.